app/helpers/api_helper.rb

```diff
@@ -0,0 +1,41 @@
+module ApiHelper
+  # Shape for a paginated response
+  def paginated(json, collection)
+    json.data do
+      yield
+    end
+    json.links do
+      json.pages! collection, url: request.original_url
+    end
+    json.meta do
+      json.count collection.total_count
+    end
+  end
+
+  # Shape for an object
+  def obj(json, object)
+    json.id object.hashid
+    json.type object.class.try(:api_type) || object.class.name.underscore
```

Comment:
The route should always describe what should be returned.

Reply:
Notes from meeting:

To address the previous comment, `type`s are helpful in the case of nested objects. For example,
```json
{
  "id": "dNuEgaE3N",
  "type": "hackathon",
  "name": "Assemble",
  "subscribers": [
    {
      "id": "sl3jsdfLs",
      "type": "user",
      "name": "Gary Tou"
    }
  ]
}
```

Using class names for `type` strongly couples the API type to the backend implementation. Ideally, the API is decoupled from the Rails model implementation. This way, if we need to refactor the models, the API can still provide consistent types without breaking changes. Also, as with many fields intended for programmatic use, they are lowercase (although there isn't an exact standard around this).

We had discussed three possible implementations:
- `object.class.name.underscore`
	- `Hackathon` -> `hackathon`
	- `Hackathon::Digest` -> `hackathon/digest`
	- `Hackathon::ExampleClass` -> `hackathon/example_class`
	- Pros
		- Preserves namespace vs class name (also a con)
	- Cons
		- Preserving namespace vs class somewhat couples implementation
		- The slash is kinda weird/ugly
- `object.class.name.underscore.parameterize`
	- `Hackathon` -> `hackathon`
	- `Hackathon::Digest` -> `hackathon-digest`
	- `Hackathon::ExampleClass` -> `hackathon-example_class`
	- Pros
		- This is how gems are named! `-` are modules and `_` separate pascal case
	- Cons
		- Preserving namespace vs class somewhat couples implementation
- `object.class.name.underscore.gsub("/", "_")`
	- `Hackathon` -> `hackathon`
	- `Hackathon::Digest` -> `hackathon_digest`
	- `Hackathon::ExampleClass` -> `hackathon_example_class`
	- Pros
		- It's clean lookin' 🤷‍♂️
	- Cons
		- Chance of unintentional `type` conflict. For example, `Hackathon::ExampleClass` and `HackathonExample::Class` have the same `type`.

**We'll be going with `object.class.name.underscore.parameterize`**

Reply:
[`c8458ed` (#57)](https://github.com/hackclub/hackathons-backend/pull/57/commits/c8458ed8a52271c56aa7c6408511c2cf3fb2b261)